StripeFinancialConnections/StripeFinancialConnections/Source/Web/FinancialConnectionsWebFlowViewController.swift

```diff
@@ -424,30 +425,75 @@ private extension FinancialConnectionsWebFlowViewController {
+                parameters.append("link_mode=\(linkMode.rawValue)")
+            }
+        }
+
+        if let prefillDetails = prefillDetails {
+            if let email = prefillDetails.email, !email.isEmpty {
+                parameters.append("email=\(email)")
+            }
+            if let phoneNumber = prefillDetails.unformattedPhoneNumber, !phoneNumber.isEmpty {
+                parameters.append("linkMobilePhone=\(phoneNumber)")
+            }
+            if let countryCode = prefillDetails.countryCode, !countryCode.isEmpty {
+                parameters.append("linkMobilePhoneCountry=\(countryCode)")
             }
         }
-        if let prefillDetails {
-            if let email = prefillDetails.email {
-                additionalQueryParameters = additionalQueryParameters + "&email=\(email)"
+
+        if let billingDetails = billingDetails {
+            if let name = billingDetails.name, !name.isEmpty {
+                parameters.append("billingDetails[name]=\(name)")
             }
-            if let phoneNumber = prefillDetails.unformattedPhoneNumber {
-                additionalQueryParameters = additionalQueryParameters + "&linkMobilePhone=\(phoneNumber)"
+            if let email = billingDetails.email, !email.isEmpty {
+                parameters.append("billingDetails[email]=\(email)")
             }
-            if let countryCode = prefillDetails.countryCode {
-                additionalQueryParameters = additionalQueryParameters + "&linkMobilePhoneCountry=\(countryCode)"
+            if let phone = billingDetails.phone, !phone.isEmpty {
+                parameters.append("billingDetails[phone]=\(phone)")
             }
+            if let address = billingDetails.address {
+                if let city = address.city, !city.isEmpty {
+                    parameters.append("billingDetails[address][city]=\(city)")
+                }
+                if let country = address.country, !country.isEmpty {
+                    parameters.append("billingDetails[address][country]=\(country)")
+                }
+                if let line1 = address.line1, !line1.isEmpty {
+                    parameters.append("billingDetails[address][line1]=\(line1)")
+                }
+                if let line2 = address.line2, !line2.isEmpty {
+                    parameters.append("billingDetails[address][line2]=\(line2)")
+                }
+                if let postalCode = address.postalCode, !postalCode.isEmpty {
+                    parameters.append("billingDetails[address][postal_code]=\(postalCode)")
+                }
+                if let state = address.state, !state.isEmpty {
+                    parameters.append("billingDetails[address][state]=\(state)")
+                }
+            }
+        }
+
+        // Join all values with an &, and URL encode.
+        guard let result = parameters.joined(separator: "&").addingPercentEncoding(withAllowedCharacters: .urlHostAllowed) else {
```

Comment:
Nit: Can you add a comment about why the URL encoding is necessary here?

Reply:
Done in https://github.com/stripe/stripe-ios/pull/4203/commits/04ff788a61c1b4a2eda6e782d05b75def9452af5